### PR TITLE
add support for pgsql filter & modules

### PR DESCRIPTION
**Pull request by hkadakia**
This PR adds support for using filters in PgSQL. It also moves some of the code from grafeas to this repo & fixes some bugs in the process.

### CODE REVIEW

**Comment by hkadakia** [comment]
@aysylu @wkozlik Please help review.

File: go/v1beta1/storage/filter.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+	return "NO SQL"
+
+}
+
+// ParseFilter parses the incoming filter and returns a formatted SQL query.
+func (fs *FilterSQL) ParseFilter(filter string) string {
+	log.Println(filter)
+	// replace string value for kind with integer
+	/*
```
Comment (aysylu) [inline go/v1beta1/storage/filter.go]: Is this meant as testing code?

File: go/v1beta1/storage/filter.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+			// log.Println(filter)
+		}
+	*/
+	s := common.NewStringSource(filter, "urlParam") // function
+	result, err := parser.Parse(s)
+	if err != nil {
+		log.Println(err)
+		return ""
+	}
+	// log.Println(result.Expr.GetCallExpr().Args[1].GetCallExpr().Args[0].GetIdentExpr().Name)
```
Comment (aysylu) [inline go/v1beta1/storage/filter.go]: could you please remove the debugging statements that are no longer needed?

**Review by aysylu — CHANGES_REQUESTED** [review]
This is very cool! Couple minor pieces of feedback.

**Review by aysylu — APPROVED** [review]

**Comment by aysylu** [comment]
Thanks for this great contribution, @hkadakia!

**Pull request merged**